src/Arcus.Messaging.Tests.Unit/Fixture/Order.cs

```diff
@@ -7,10 +7,69 @@ namespace Arcus.Messaging.Tests.Unit.Fixture
     /// <summary>
     /// Test message implementation to test specific message handling scenario's.
     /// </summary>
-    public class Order
+    public class Order : IEquatable<Order>
```

Comment:
If you specify `Order` to be a `record` instead of a `class`, you do not have to worry anymore about implementing IEquatable, Equals, etc... :)

Reply:
Ah, we could do that, indeed. (We could also use F# 😄 ).

Reply:
That it is a test project doesn't really matter here, as it was about the general Arcus contribution that I was talking about 😄 . It is a way of learning, that's true.
Let me check if it reacts as expected with the other tests, as it is used quite a lot.

Reply:
Ah, oops, we test both .NET Core and .NET 6 in the unit tests, so it would only work for .NET 6. We could however come back to this if we remove .NET Core from Arcus.

Reply:
Is it ?  Isn't it a language feature that is not coupled with a .NET version ?